storage/memoized_iterator.go

```diff
@@ -101,29 +100,26 @@ func (b *MemoizedSeriesIterator) Seek(t int64) chunkenc.ValueType {
 }
 
 // Next advances the iterator to the next element.
```

Comment:
Perhaps add a note why `next` is un-exported and what the difference in behavior is.

Reply:
With your note in, I now get second thoughts about the unexporting. I think `next` behaves fine and isn't violating any contract of the original `Next`. The specifics about the previous samples are really specific to the `MemoizedSeriesIterator`. Plus, we aren't even implementing a normal `chunkenc.Iterator` anyway (because other methods got removed). I now think we can just leave `Next` as Next`, maybe documenting that the `PeekBack` thing only works with `Next` and not with `Seek`. @codesome WDYT?

Reply:
I agree that it didn't violate the contract of the original `next`. I was thinking the reason to not export it is because it might be surprising that it doesn't work correctly with `PeekBack` but just having that comment could also work.

Reply:
Isn't it actually the other way around? _Only_ after a call of `Next` (or `next` ;) `PeekBack` will work correctly. After `Seek`, `PeekBack` was never supposed to work.

And that behavior didn't change with this PR, did it?

I would really make `next` `Next` again, and if at all, add to the doc comment of `Seek` that, after it, `PeekBack` will return false.

Reply:
Yeah, PeekPrev is only ever used after Seek. Since Seek uses Next internally, the current behavior might be needed.

This iterator has a weird semantics… I think indeed we better don't touch anything here. A more comprehensive explanation might be good at some time in the future so that not everyone has to wrap their head around this semantics again.